Add Smart Playlist plugin documentation (#677)
music-assistant/music-assistant.io: src/content/docs/plugins/smart_playlist.md

---
title: Smart Playlists
description: Create dynamic or fixed playlists based on rules like genres, artists, albums, favorites and similar tracks.
---

> import { Aside } from '@astrojs/starlight/components';

Reviewer: This isnt working in the preview so not sure what this is for? I can see you have used it below but that isnt how CAUTIONS are normally done?

# Smart Playlists

> The Smart Playlists plugin lets you create rule-based playlists from your Music Assistant library. Playlists can be **Dynamic** — tracks are re-evaluated fresh every time you play them — or **Fixed (one-time)**, generating a regular editable playlist once from your current library.

Reviewer: ```suggestion
The Smart Playlists plugin lets you create rule-based playlists from your Music Assistant library. Playlists can be **Dynamic** whereby tracks are re-evaluated fresh every time you play them or **Fixed (one-time)**, which generates an editable playlist from your current library.
```

<Aside type="caution" title="Experimental">
This plugin is currently marked as **experimental**. Behaviour may change before the final release.
</Aside>

## Installation

Enable the **Smart Playlists** plugin in **Settings → Providers → Add Provider → Plugins**. No additional configuration is required.

![Smart Playlists plugin entry](/assets/screenshots/smart_playlist/add_plugin.png)

## Creating a Smart Playlist

> In the Playlists view, select the **⊕** menu in the top right and choose **Create Smart Playlist**.

Reviewer: ~~So this isnt just going to be another option in here~~

Actually I just looked at the images. The icon you use here isnt correct. In these cases I normally take a screenshot of the actual icon, make it a PNG and then inline it in the text at an appropriate size.

![Create Smart Playlist menu](/assets/screenshots/smart_playlist/menu_entry.png)

![Create Smart Playlist dialog](/assets/screenshots/smart_playlist/create_smart_playlist.png)

Enter a name and choose a mode:

> - **Dynamic** — Rules are re-evaluated every time the playlist is played. The playlist always reflects your current library.

Reviewer: ```suggestion
- **Dynamic.** Rules are re-evaluated every time the playlist is played. The playlist is always derived from your current library.
```

> - **Fixed (one-time)** — Rules are evaluated once at creation time. The result is saved as a normal editable playlist. When **Fixed** is selected, a **Number of tracks** stepper appears to set the exact track count.

Reviewer: ```suggestion
- **Fixed (one-time).** Rules are evaluated once at creation time. The result is saved as a normal editable playlist. When **Fixed** is selected, a **Number of tracks** stepper appears to set the exact track count.
```

![Fixed (one-time) playlist dialog](/assets/screenshots/smart_playlist/create_fixed_playlist.png)

> The dialog is divided into four collapsible sections.

Reviewer: ```suggestion
The dialog is divided into four collapsible sections; Source, Artists & Albums, Filters, and Advanced.
```

**Note:** Only one of *Similar to track* or *Similar to artist* can be active at the same time. Both fields require a music provider that supports similar-track or similar-artist recommendations; if no such provider is configured, the fields are disabled.

## Artists & Albums

![Artists & Albums section expanded](/assets/screenshots/smart_playlist/show_artists_albums.png)

| Option | Description |
|--------|-------------|
| **Artists** | Limit tracks to those by the selected artists. |
| **Albums** | Limit tracks to those from the selected albums. |
| **Exclude artists** | Remove all tracks by the selected artists from the results. |
| **Exclude albums** | Remove all tracks from the selected albums from the results. |

## Filters

![Filters section expanded](/assets/screenshots/smart_playlist/show_filters.png)

| Option | Description |
|--------|-------------|
| **Favorites only** | When enabled, only tracks marked as favorites are included. |
| **Minimum popularity** | Slider from 0–100. Only tracks with a popularity score at or above the threshold are included. Requires a provider that supplies popularity data (e.g. Spotify, Tidal). Set to *Any* to disable. |

> | **Year range** | Enter a *From year* and/or *To year* to limit tracks by release year. |

Reviewer: This probably should be on the frontend or server PR but I dont see release_date populated at track level for anything in my database? You may ve sourcing this from the album but just thought I would check

## Advanced

![Advanced section expanded](/assets/screenshots/smart_playlist/show_advanced.png)

| Option | Description |
|--------|-------------|

> | **No repeat within** | Slider in hours (1–8760). Tracks played within this time window are excluded from the next evaluation, preventing repeats. Set to *Off* to disable. Best-effort: if the filtered library is too small to fill the playlist without repeats, the least-recently-played tracks are used to top it up; if all tracks were played recently, the dedup constraint is dropped entirely so the playlist never goes empty. |

Reviewer: ```suggestion
| **No repeat within** | Slider in hours (1–8760). Tracks played within this time window are excluded from the next evaluation, preventing repeats. Set to *Off* to disable. Best-effort: if the filtered library is too small to fill the playlist without repeats, the least-recently-played tracks are used to top it up; if all tracks were played recently, the de-duplication constraint is dropped entirely so the playlist never goes empty. |
```

| **Rule combination** | **Match ALL rules (AND)** — a track must satisfy every active rule. **Match ANY rule (OR)** — a track only needs to satisfy one rule. |

## Track count indicator

> While configuring rules, the dialog shows the approximate number of matching tracks at the bottom (e.g. *~2501 Titel*). This updates live as you change the rules so you can tune the playlist before saving.

Reviewer: ```suggestion
While configuring rules, the dialog shows the approximate number of matching tracks at the bottom (e.g. *~100 Tracks*). This updates live as you change the rules so you can tune the playlist before saving.
```

![Track count indicator](/assets/screenshots/smart_playlist/track_count.png)

## Notes

- Smart playlists only exist as long as the Smart Playlists plugin is installed. Removing the plugin also removes all associated smart playlists from the MA library.
- Renaming a smart playlist in the MA library is automatically reflected in the stored rules.
- *Similar to track* and *Similar to artist* are mutually exclusive — only one can be set per playlist.

> - The **Minimum popularity** filter only works if at least one of your music providers supplies popularity metadata.

Reviewer: These two points are duplicating text further up the page and thus can be removed
